primitives/staking/src/lib.rs

```diff
@@ -37,6 +37,22 @@ pub type SessionIndex = u32;
 /// Counter for the number of eras that have passed.
 pub type EraIndex = u32;
 
+/// Representation of a staking account, which may be a stash or controller account.
+///
+/// Note: once the controller is completely deprecated, this enum can also be deprecated in favor of
+/// the stash account.
+#[derive(Copy, Clone, Debug)]
+pub enum StakingAccount<AccountId> {
```

Comment:
I am assuming this is not stored anywhere in storage given the lack of derives, which is good.

Don't wanna be picky, but `Copy` for account id is something that we typically don't 🙉

Reply:
removed